# Getting chemical fingerprints
## Find its ID and then download molfile

In [2]:
from bioservices import chebi
#import libchebipy

instance = chebi.ChEBI()

#open file
f = open('drugs.txt')
druglist = []
for lines in f:
    druglist.append(lines)
f.close()

nameList = {} #keys = drug name; value = idList
# https://pythonhosted.org/bioservices/_modules/bioservices/chebi.html
# http://libchebi.github.io/libChEBI%20API.pdf
molfile = open('alldrug.sdf', 'w')
for drug in druglist:
    result = instance.getLiteEntity(drug)
    if len(result) == 0:
        print(drug)
        nameList[drug] = {}
    else:
        idList = {} #keys = query name; value = ID
        
        for query in range(len(result)):
            idList[result[query][1]] = result[query][0] #save id in idList
            realID = result[query][0][6:]
            #opt out to get molfile and write to file named with its ID
            StructureInfo = instance.getUpdatedPolymer(realID)
            if StructureInfo != None:
                print(realID, file = molfile)
                print(StructureInfo['updatedStructure'], file = molfile)
                print('>  <ACTIVITY>\n5.00\n$$$$',file=molfile)
            else:
                print(realID, 'None')
        nameList[drug] = idList

molfile.close()

       



ampicillin-sulbactam

85989 None
17833 None
5880 None
piperacillin-tazobactam

3770 None
35006 None
75782 None
79088 None
75769 None
60145 None
59062 None
133022 None
136651 None
64932 None
78681 None
75047 None
67222 None
38551 None
5022 None
5034 None
5057 None
ticarcillin/clavulanate



## some ID does not have a structure: THE ABOVE `some_ID None`
they are mostly *unrelated structure* or *general structure*, some of them are *mixtures*
### Detailed List:
85989 None: mixture

17833 None:group

5880 None: mixture

3770 None: mixture: co-trimazazole

35006 None: is an organic molecule entity ?

75782 None: Vitamin B complex (WT*?)

79088 None: EC 3.4.24.35 (gelatinase B) inhibitor

75769 None: B vitamin (WT* again)

60145 None: Poly[B-(methylamino)borazine] polymer is a homopolymer, composed of poly[B-(methylamino)borazine] macromolecules.

59062 None: Polymyxins are antibiotics with a general structure...

133022 None: B-cell lymphoma 2 inhibitor

136651 None: S100 calcium-binding protein B inhibitor

64932 None: cathepsin B inhibitor

78681 None: tropomyosin-related kinase B receptor antagonist

75047 None: E.coli metabolite

67222 None: not related compound

38551 None: Cytochrome B

5022 None: cytochrome again

5034 None: cytochrome again

5057 None: Fibrinopeptide B
### This suggest we should revisit the search results since so many garbages are inside...

In [12]:
instance.getUpdatedPolymer('85989') == None





True

## Take a closer look at the search result `nameList`

In [14]:
nameList


{'amikacin\n': {amikacin: CHEBI:2637,
  amikacin disulfate: CHEBI:2638,
  amikacin(4+): CHEBI:84739},
 'ampicillin\n': {N-benzoylampicillin: CHEBI:63221,
  ampicillanyl group: CHEBI:53713,
  ampicillin: CHEBI:28971,
  ampicillin benzathine: CHEBI:51343,
  ampicillin potassium: CHEBI:51342,
  ampicillin sodium: CHEBI:34535,
  ampicillin trihydrate: CHEBI:31209,
  ampicillin(1-): CHEBI:50658,
  ampicilloyl group: CHEBI:53704,
  ampicilloyl-butylamine: CHEBI:55473,
  bacampicillin: CHEBI:2968,
  pivampicillin: CHEBI:8255},
 'ampicillin-sulbactam\n': {},
 'aztreonam\n': {SQ 26,992: CHEBI:62665,
  aztreonam: CHEBI:161680,
  aztreonyl group: CHEBI:60429},
 'carbapenem\n': {3-[(2-aminoethyl)sulfanyl]-6-ethyl-7-oxo-1-azabicyclo[3.2.0]heptane-2-carboxylic acid: CHEBI:132289,
  3-[(2-aminoethyl)sulfanyl]-6-methyl-7-oxo-1-azabicyclo[3.2.0]heptane-2-carboxylic acid: CHEBI:132290,
  3-[(2-aminoethyl)sulfanyl]-7-oxo-1-azabicyclo[3.2.0]heptane-2-carboxylic acid: CHEBI:132304,
  carbapenem: CHEBI:4676

In [14]:
instance.getUpdatedPolymer(28417)

(UpdatedPolymer){
   updatedStructure = " 
 
 
 34 36  0  0  1  0  0  0  0  0999 V2000
   15.9600  -21.4900    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   15.9600  -22.8900    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   17.1724  -23.5900    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   18.3849  -22.8900    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   18.3849  -21.4900    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   17.1724  -20.7900    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
   19.6160  -20.7790    0.0000 O   0  0  0  0  0  0  0  0  0  0  0  0
   20.8284  -20.0790    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   22.0692  -20.7955    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   23.2817  -20.0956    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   23.2819  -18.6956    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   22.0411  -17.9791    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   20.8286  -18.6790    0.0000 C   0  0  0  0  0  0  0  0  0  0  0  0
   

# Q1: What if it is a combination of drugs? (penicillin + beta-lactamase inhibitor)
# Q2: does charge affect TMACC descriptor?

`scp mofile hermuba@192.168.56.1`->tmacc.jar->get 